ocp_resources/cluster_role.py

```diff
@@ -33,19 +33,46 @@ def __init__(
         self.api_groups = api_groups
         self.permissions_to_resources = permissions_to_resources
         self.verbs = verbs
+        self.desired_state = {"rules": []}
 
     def to_dict(self):
-        res = super().to_dict()
+        self.res = super().to_dict()
```

Comment:
if you set a value to setf.res here, you need to define it in __init__, do not hide it here

Reply:
Defined in Resource.init()

Reply:
ack